Rework ReceiveAndPlot

tstenner opened the pull request:
Improve ReceiveAndPlot somewhat:

- use a preallocated buffer for pull_chunk (so even an ancient laptop can show 2 channels @100KHz)
- decouple scrolling and pulling for a smoother experience
- allow multiple data and marker streams
- display all suitable (i.e. data or marker streams) streams found at start time instead of one data and one marker stream

cboulay commented:
This is just an example of how to use pylsl. Defining a new class called `Inlet` is confusing. In general, I think we should keep OOP out of these examples because it'll scare away beginners.

If you want a good tool for pulling and plotting data then that's great, I think it's sorely lacking. I'd love to get rid of the Matlab visualizer as the default demo app. App-SigVisualizer is a good start but it needs some work. I think Clemens has a different solution we should also look at.

tstenner commented:
> This is just an example of how to use pylsl.

And the extended version is below 100 LOC.

> Defining a new class called `Inlet` is confusing.

Regarding the name I agree, so I renamed it.
Handling different inlet types differently with inheritance isn't that complicated, especially when compared to several for-loops.

> In general, I think we should keep OOP out of these examples because it'll scare away beginners.

For the basic examples I agree, but plotting signals in realtime is something for advanced users.

I added some more comments and docstrings so it's easier to understand.

> If you want a good tool for pulling and plotting data then that's great, I think it's sorely lacking.

I want that, but in a pinch I'd also take a quick and dirty `python -m pylsl.examples.ReceiveAndPlot` that 

> I'd love to get rid of the Matlab visualizer as the default demo app. App-SigVisualizer is a good start but it needs some work. I think Clemens has a different solution we should also look at.

We're using lsl in the clinic, so we can neither justify buying academic licenses for Matlab nor spending $3000+ for a regular license. For a fully featured visualization tool the SigVisualizer would be my first chioce, but right now it's too complicated to use.
This example shouldn't be the de facto visualizer, but I think it hits a sweet spot between a usable visualizer and API example.
